.github/workflows/pypi-whl.yml

```diff
@@ -0,0 +1,96 @@
+name: PyPI Whl Publish
+
+on:
+  push:
+    branches:
+      - main
+      - deno-binary
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build_wheels:
+    name: Build whl on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # windows-latest is an Intel runner
+        # macos-13 is an Intel runner, macos-latest is an Apple Silicon runner
+        # ubuntu-latest is an Intel runner, ubuntu-24.04-arm is an arm runner
+        os: 
+          - windows-latest
+          - macos-13
+          - macos-latest
+          - ubuntu-latest
+          - ubuntu-24.04-arm
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: denoland/setup-deno@v2
+        with:
+          deno-version: v2.x
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
+      - name: Install build and twine
+        run: python -m pip install build twine
+      - name: Compile deno binaries
+        run: |
+          cd tools/pypi-whl
+          deno compile -ERNW --allow-run -o dist/bids-validator ../../src/bids-validator.ts
+      - name: Print out the content of the dist folder after building validator binary
+        run: ls -l tools/pypi-whl/dist/
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bids-validator-${{ matrix.os }}
+          path: |
+            tools/pypi-whl/dist/bids-validator
+            tools/pypi-whl/dist/bids-validator.*
+      - name: Build wheel
+        run: python -m build --wheel
+        working-directory: tools/pypi-whl
+      - name: Print out the content of the dist folder after building the wheel
+        run: ls -l tools/pypi-whl/dist/
+      - name: Validate the wheel installs the BIDS validator correctly
+        run: |
+          python -m venv .venv
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            ACTIVATE_PATH=.venv/Scripts/activate
+            BIDS_VALIDATOR_BINARY_NAME=bids-validator.exe
+            VENV_BIN_DIR=.venv/Scripts
+          else
+            ACTIVATE_PATH=.venv/bin/activate
+            BIDS_VALIDATOR_BINARY_NAME=bids-validator
+            VENV_BIN_DIR=.venv/bin
+          fi
+          source "$ACTIVATE_PATH"
+          pip install tools/pypi-whl/dist/*.whl
+          bids-validator --version
+          ls -lh "$VENV_BIN_DIR"
+          diff "$VENV_BIN_DIR"/"$BIDS_VALIDATOR_BINARY_NAME" tools/pypi-whl/dist/"$BIDS_VALIDATOR_BINARY_NAME"
+          deactivate
```

Comment:
move this into a helper script e.g. `tools/pypi-whl/test-installation` -- too much for the workflow - makes it hard to read.

Reply:
done